### PR TITLE
[MPLUGIN-427] Expose generics information of parameter types in report

### PR DESCRIPTION
**Pull request by kwin**
and help mojo

### CODE REVIEW

File: maven-plugin-tools-annotations/src/test/java/org/apache/maven/tools/plugin/extractor/annotations/scanner/DefaultMojoAnnotationsScannerTest.java [hunk 1]
```diff
@@ -76,4 +80,48 @@ void scanDeprecatedMojoAnnotatins() throws ExtractionException, IOException
...
+
+        ParameterAnnotationContent parameter = annotatedClass.getParameters().get( "string" );
+        assertNotNull( parameter );
+        assertEquals( "java.lang.String", parameter.getClassName() );
+        assertThat( parameter.getTypeParameters() ).isEmpty();
+
+        parameter = annotatedClass.getParameters().get( "stringBooleanMap" );
+        assertNotNull( parameter );
+        assertEquals( "java.util.Map", parameter.getClassName() );
+        assertThat( parameter.getTypeParameters() ).contains( "java.lang.String", "java.lang.Boolean" );
```
[the remark this answers is not in the document]
Comment (kwin) [reply maven-plugin-tools-annotations/src/test/java/org/apache/maven/tools/plugin/extractor/annotations/scanner/DefaultMojoAnnotationsScannerTest.java]: Done in https://github.com/apache/maven-plugin-tools/pull/159/commits/4816ac73c9f622a7122c56ee6b478c39b6bc7211. I am so used to using Hamcrest, where `contains` has other semantics.

**Review by slawekjaranowski — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
@kwin 

One issue what I see: in `target/classes/META-INF/maven/plugin.xml` is:

```
  <mojos>
    <mojo>
        <parameter>
          <name>reactorProjects</name>
          <type>java.util.List</type>
          <required>true</required>
          <editable>false</editable>
          <description></description>
        </parameter>
....

      <configuration>
     ...
     <reactorProjects implementation="java.util.List&lt;org.apache.maven.project.MavenProject&gt;" default-value="${reactorProjects}"/>
   ...
      </configuration>
...
```

should we have `parameter type` in `configuration`

**Comment by kwin** [comment]
@gnodet Good catch, will make sure to emit the simple type there as well.

**Comment by jira-importer** [comment]
Resolve #741

**Pull request merged**